src/api/ApiCall.hh

```diff
@@ -11,21 +11,62 @@ namespace ra {
 namespace api {
 
 enum class ApiResult
-{    
+{
     None = 0,       // unspecified
     Success,        // call was successful
     Error,          // an error occurred
     Failed,         // call was unsuccessful, but didn't error
     Unsupported,    // call is not supported and was not made
+    Incomplete,     // call could not be made at this time
 };
 
 struct ApiRequestBase
 {
+protected:
     template<class TRequest, class TCallback>
     static void CallAsync(const TRequest& request, TCallback&& callback)
     {
         ra::services::ServiceLocator::GetMutable<ra::services::IThreadPool>().RunAsync(
-            [request, callback = std::move(callback)] { callback(request.Call()); });
+            [request, callback = std::move(callback)]{ callback(request.Call()); });
+    }
+
+    template<class TRequest, class TCallback>
+    static void CallAsyncWithRetry(const TRequest& request, TCallback&& callback)
+    {
+        ra::services::ServiceLocator::GetMutable<ra::services::IThreadPool>().RunAsync(
+            [request, callback = std::move(callback)]
+        {
+            DoAsyncWithRetry(std::move(request), std::move(callback), std::chrono::milliseconds(0));
+        });
+    }
+
+private:
+    template<class TRequest, class TCallback>
+    static void DoAsyncWithRetry(const TRequest&& request, TCallback&& callback, std::chrono::milliseconds delay)
```

Comment:
Does request really have to be const like that? I'm going to suggest to get rid of the `const`, or remove one of the `&`. A const temporary doesn't make any sense.